### PR TITLE
Fixes MODsuit DNA scanner runtimes

### PR DESCRIPTION
**Pull request by Arturlang**
## About The Pull Request
The modsuit DNA lock seemed to lack sanity and only cared about whoever wore it, even though it's called when manipulating the modules inside the suit.
This PR deals with that and fixes the action buttons so they actually work, instead of just generating a runtime

## Why It's Good For The Game
Less runtimes is always good, and fixing features that don't work is dandy

## Changelog

:cl:
fix: Modsuits DNA lock module will no longer stop you from changing up the suit's modules while not worn, even if you are it's rightfull owner.
/:cl:

### CODE REVIEW

File: code/modules/mod/modules/modules_general.dm [hunk 1]
```diff
@@ -571,10 +571,13 @@
 	. = ..()
 	on_emag(src, user, emag_card)
 
-/obj/item/mod/module/dna_lock/proc/dna_check()
-	if(!dna || (mod.wearer.has_dna() && mod.wearer.dna.unique_enzymes == dna))
+/obj/item/mod/module/dna_lock/proc/dna_check(mob/user)
+	if(!user && !iscarbon(user))
+		return FALSE
```
Comment (MrMelbert) [inline code/modules/mod/modules/modules_general.dm:576]:
```suggestion
	if(!iscarbon(user))
		return FALSE
```
`!iscarbon()` checks for nulls, so no need to check for a missing user.

File: code/modules/mod/modules/modules_general.dm [hunk 2]
```diff
@@ -571,10 +571,13 @@
...
+	var/mob/living/carbon/C = user
+	if(!dna  || (C.has_dna() && C.dna.unique_enzymes == dna))
 		return TRUE
-	balloon_alert(mod.wearer, "dna locked!")
+	balloon_alert(user, "dna locked!")
```
Comment (MrMelbert) [inline code/modules/mod/modules/modules_general.dm:580]: Better var name here. something like `carbon_user` instead of `C`

**Review by MrMelbert — CHANGES_REQUESTED** [review]
Can't comment on the fix itself, since I don't know much of MODsuit code yet, but a few code tweaks

File: code/modules/mod/modules/modules_general.dm (hunk 2 above)
Comment (Arturlang) [reply code/modules/mod/modules/modules_general.dm:580]: Names are hard, suggestions put in

**Comment by Arturlang** [comment]
Ugh, not going to bother with fixing the GPG signing for this PR.

File: code/modules/mod/mod_actions.dm [hunk 3]
```diff
@@ -97,7 +97,7 @@
 	var/mob/pinner
 
 /datum/action/item_action/mod/pinned_module/New(Target, obj/item/mod/module/linked_module, mob/user)
-	if(user == mod.ai)
+	if(user == mod?.ai)
```
Comment (Fikou) [inline code/modules/mod/mod_actions.dm:100]: done in #63791

**Review by Fikou — APPROVED** [review]
dna fix looks good

**Comment by Arturlang** [comment]
Managed to verify the commit, just a lot of annoying parts in non-obvious ways making the stuff shitty to figure out.
Only took me 6 force pushes.

File: code/modules/mod/mod_actions.dm (hunk 3 above)
Comment (Arturlang) [reply code/modules/mod/mod_actions.dm]: Removed from my PR

**Pull request merged**